newapi/ooniapi/measurements.py

```diff
@@ -284,38 +293,39 @@ def _fetch_autoclaved_measurement_body(report_id: str, input) -> Optional[bytes]
         current_app.logger.error(f"missing autoclaved for {report_id} {input}")
         # This is a bug somewhere: the msmt is is the fastpath table and
         # not in the autoclavedlookup table
-        return None
+        raise MsmtNotFound
 
     body = _fetch_autoclaved_measurement_body_from_s3(
         r.filename, r.frame_off, r.frame_size, r.intra_off, r.intra_size
     )
     return body
 
 
-def _fetch_measurement_body(report_id, input: str) -> Optional[bytes]:
+def _fetch_measurement_body(report_id, input: str, measurement_uid) -> bytes:
     """Fetch measurement body from either disk, jsonl or autoclaved on S3"""
     log.debug(f"Fetching body for {report_id} {input}")
     u_count = report_id.count("_")
-    # The number of underscores can be:
-    # 1: Legacy measurement e.g.
-    # 20141101T220015Z_OzDkiPoJMVjHItj<redacted>
-    # 2: Legacy measurement e.g.
-    # 20201019T050625Z_AS8369_VVqkepmw<redacted>
     # 5: Current format e.g.
     # 20210124T210009Z_webconnectivity_VE_22313_n1_Ojb<redacted>
     if u_count == 5:
         # Look on disk and then from JSONL cans on S3
         body = _fetch_measurement_body_on_disk(report_id, input)
```

Comment:
Maybe for consistency this function should also `raise MsmtNotFound` instead of checking for the body being None.

Reply:
Not finding the msmt on disk is not an error. It's expected to happen often.